backend/routes/users.js

```diff
@@ -0,0 +1,127 @@
+import express from "express";
+
+import { User } from "../models/userSchema";
+import { authenticateUser } from "../middlewares/authenticateUser";
+
+const router = express.Router();
+
+//User Endpoints
+
+router.post("/register", async (req, res) => {
+  try {
+    const {
+      firstname,
+      lastname,
+      email,
+      address,
+      password,
+      allergies,
+      pros,
+      recommendedfor,
+    } = req.body;
+
+    if (!firstname) {
```

Comment:
I like how you did the If statements, very clean to have a ! and then the error message instead of loads of nested If-statements! 

Nice!